src/lib.rs

```diff
@@ -251,11 +251,17 @@ impl<'t> MachineDescriptor<'t> {
         let sess = sess.connect(&self.public_ip).await?;
         tracing::trace!("connected");
 
+        let public_ip = self.public_ip;
+        // if not defined, set public dns to be the public ip
+        let public_dns = self
+            .public_dns
+            .map(String::from)
```

Comment:
Why do you need this `map`? Isn't `public_dns` already `Option<String>`?

Reply:
Ah yes. Bad copy-paste 😅